src/FffGcodeWriter.cpp

```diff
@@ -1211,6 +1211,356 @@ void FffGcodeWriter::processSpiralizedWall(const SliceDataStorage& storage, Laye
     gcode_layer.spiralizeWallSlice(&mesh_config.inset0_config, wall_outline, last_wall_outline, seam_vertex_idx, last_seam_vertex_idx);
 }
 
+static bool polysIntersect(const Polygons& poly_a, const Polygons &poly_b)
+{
+    // only do the full intersection when the polys' BBs overlap
+    AABB bba(poly_a);
+    AABB bbb(poly_b);
+    return bba.hit(bbb) && poly_a.intersection(poly_b).size() > 0;
+}
+
+static bool polyOutlinesAdjacent(const ConstPolygonRef inner_poly, const ConstPolygonRef outer_poly, const coord_t max_gap)
+{
+    const coord_t max_gap2 = max_gap * max_gap;
+    const unsigned outer_poly_size = outer_poly.size();
+    for (unsigned line_index = 0; line_index < outer_poly_size; ++line_index)
+    {
+        const Point lp0 = outer_poly[line_index];
+        const Point lp1 = outer_poly[(line_index + 1) % outer_poly_size];
+        for (Point inner_poly_point : inner_poly)
+        {
+            if (LinearAlg2D::getDist2FromLineSegment(lp0, inner_poly_point, lp1) < max_gap2)
+            {
+                return true;
+            }
+        }
+    }
+    return false;
+}
+
+static int findAdjacentEnclosingPoly(const ConstPolygonRef& enclosed_inset, const std::vector<ConstPolygonRef>& possible_enclosing_polys, const coord_t max_gap)
```

Comment:
These functions should be moved to utils/polygonUtils.cpp

Also you might want to have a look at the functions in that file, because there is some similar functionality in there.